src/integration-tests/test_poison_messages.py

```diff
@@ -443,7 +443,7 @@ def test_poison_proxy_and_replica_fanout(self, multi_node: Cluster):
             multi_node, proxy, tc.DOMAIN_FANOUT, ["?id=foo", "?id=bar", "?id=baz"]
         )
 
-    @max_delivery_attempts(2)
+    @max_delivery_attempts(3)
```

Comment:
It would be good to have a note in the commit message body about why we needed this change, since it was quite subtle.